### ISSUE TITLE
Upload p[roblems in Arduino Ide V1.89

### ISSUE THREAD

**Issue by RSZ-Nld** [paulh002/VFO-ESP32-Si5351#4]
Hello Paul,

Made the version 3 kicad new, but i can not upload the files, got all sorts of problems.
Do you have  a bin file for me? Think that will be a lot easier.

Greetings

Rolf

**Comment by paulh002** [comment]
Hello,

I don't know exactly what you have built, but to build a bin I need some
more information.
Did you build the version with the TCA 9548?  Because for that one I did
not share the software yet.
The version in github is older and relies on older versions of libraries
(included in the library directory).
I have an updated version but not yet shared, because some screens are not
yet fully migrated to the latest version of lvgl v8.x

Regards,
Paul



Op zo 25 jun 2023 om 17:33 schreef RSZ-Nld ***@***.***>:

**Comment by RSZ-Nld** [comment]
Yes i build the onme with the TCA 9548.
So i can not use that now, i have to wait for the software then.
I would like to test the the board, how can i test it ?

**Comment by RSZ-Nld** [comment]
Thanks for that, i like this project !

**Comment by paulh002** [comment]
You can find the bin files and source code in this repo https://github.com/paulh002/ESP32vfo

**Comment by RSZ-Nld** [comment]
Yes the bin file is working, but no touch ? 
Screen is fine on my 3.2 Inch

**Comment by paulh002** [comment]
Can you measure if pin 11 of the TFT connects to IO0? I had an issue in the PCB, but I thought the version on github is corrected. But touch is using IO0 and should be connected to pin 11 of the TFT (T_CS)

**Comment by RSZ-Nld** [comment]
Yes pin 11 is connected with IO0

**Comment by paulh002** [comment]
I compiled for 2.4 inch, ILI9341 what display are you using?

**Comment by paulh002** [comment]
Does not matter, do you know the driver for the 3.2 inch?

Op ma 26 jun 2023 om 14:57 schreef RSZ-Nld ***@***.***>:

**Comment by paulh002** [comment]
For what I can see it is the same driver. So it has to be a software issue. I now also have a test version running and have the same issue. I have to debug it. If I compare the code with an other project it should work. I will do some tests. You can also compile the project your self, but you need to download the libraries necessary like.

**Comment by RSZ-Nld** [comment]
Yes i try that but i did not succeed yey

**Comment by paulh002** [comment]
The new code I shared should compile with latest libraries. You can give it a try.

**Comment by RSZ-Nld** [comment]
Ok nice, i got all kind of problems like:
GuiBinarySemaphore' was not declared in this scope
and
'class RotaryEncoder' has no member named 'isConnected'

All updated to the latest libraries now, i wait for the bin file then.

**Comment by paulh002** [comment]
I uploaded new version

**Comment by RSZ-Nld** [comment]
Yes that is working, but the touch screen is upside down, so i have at the top the touch for everything now.

**Comment by RSZ-Nld** [comment]
Only the touch

**Comment by paulh002** [comment]
Ok, seems to be different for 3.2 inch. Are both X and Y swapped? Or only Y?

Op ma 26 jun 2023 om 16:31 schreef RSZ-Nld ***@***.***>:

**Comment by RSZ-Nld** [comment]
Lsb is left top and Settings is Right top

**Comment by paulh002** [comment]
Ok, changed the Y for the touch. I also compiled with Arduino indeed a lot of errors. I am not using Arduino compiler. I use Visual Studio with Visual Micro extension (you have to buy for 25 Euro) But works very good. Have to check why it does not compile with   normal Arduino IDE

**Comment by paulh002** [comment]
You should be able to compile when you delete the files smeter.cpp smeter.h
wifigui.cpp and wifigui.h
These files I did not update yet for lvgl version 8, that's why they give
problems. This is still work in progress.
Also I switched off the bandpassfilter in the code. It is using a serial
chip to communicate. Currently I use I2C for this.
But updating the smeter, should not be so difficult I already did it for an
other project I did.



Op ma 26 jun 2023 om 16:39 schreef RSZ-Nld ***@***.***>:

**Comment by RSZ-Nld** [comment]
Yes now touch is perfect, thx !   Now i have to learn how i can give a Wifi in, now it is exit when i touch it. But this is a test file not complete yet?

**Comment by RSZ-Nld** [comment]
Yes i try that now, delete that files now and i will try that

**Comment by paulh002** [comment]
If you can compile you can add wifi credentials in the ino file. Also
change R.wifi_onoff = false; to true
This is still work inprogress due to the major update of lvgl from 7 to 8.
They changed a log of calls.
I was not so active with this upgrade because of all the work on the sdr
transceiver.


Op ma 26 jun 2023 om 17:00 schreef RSZ-Nld ***@***.***>:

**Comment by RSZ-Nld** [comment]
Yes did that Ide is at work now, takes a while.
But no problem it is all a hobby and you are doing a great job !
I like to thank you, when i compile i let you know if it has worked!

**Comment by RSZ-Nld** [comment]
Problems with LVGL now in Arduino IDE, that it the problem now, so that is work for a other time

**Comment by paulh002** [comment]
You should choose version 8.3.3. in the library manager for lvgl

Op ma 26 jun 2023 om 17:14 schreef RSZ-Nld ***@***.***>:

**Comment by paulh002** [comment]
![image](https://github.com/paulh002/VFO-ESP32-Si5351/assets/35396688/12ccb963-a121-4e2f-be73-2e701a128778)
In this configuration it compiles ok

**Comment by RSZ-Nld** [comment]
Ok i will try it again tomorrow and let you know.

**Comment by RSZ-Nld** [comment]
Try it again with all new in Libraries, but no luck, see:
IO.cpp:78:14: error: 'class RotaryEncoder' has no member named 'isConnected'
  if (decoder.isConnected())
              ^~~~~~~~~~~
IO.cpp:87:10: error: 'class RotaryEncoder' has no member named 'setButtonPorts'
  decoder.setButtonPorts(5, 4);
          ^~~~~~~~~~~~~~
gui.cpp:34:8: error: 'lv_disp_draw_buf_t' does not name a type; did you mean 'lv_draw_ctx_t'?
 static lv_disp_draw_buf_t draw_buf;
        ^~~~~~~~~~~~~~~~~~
        lv_draw_ctx_t

**Comment by paulh002** [comment]
I added a library directory in github.
Please replace the RotaryEncoder source files by the one here supplied.
Also in the library directory place lv_conf.h this files configures lvgl



Op di 27 jun 2023 om 11:59 schreef RSZ-Nld ***@***.***>:

**Comment by RSZ-Nld** [comment]
Done that, now i get this, but i have lvgl V8.3 and config file in scr but this problem i have now:
gui.cpp:34:8: error: 'lv_disp_draw_buf_t' does not name a type; did you mean 'lv_draw_ctx_t'?
 static lv_disp_draw_buf_t draw_buf;
        ^~~~~~~~~~~~~~~~~~
        lv_draw_ctx_t

**Comment by paulh002** [comment]
lv_disp_draw_buf_t is defined in lv_hal_disp.h in the lvgl library directory. The only reason that you get this error is that somewhere an older version on lvgl is found by arduino IDE. Make sure to remove all old lvgl directories. Lvgl versio 7.11 did not have this typedef

**Comment by RSZ-Nld** [comment]
I think i have got it now, problem was the Arduino IDE V1.8.19 i downgrade it to V1.8.16 like you have and now it is working !
I got only a white screen, must be a mistake.
Compiling is now ok

**Comment by paulh002** [comment]
Ok, the white screen can be because of the settings in
TFT_eSPI\User_setup.h enable the #define ILI9341_DRIVER and set the pin
numbers to:
#define TFT_MISO 32
#define TFT_MOSI 25
#define TFT_SCLK 18
#define TFT_CS   5  // Chip select control pin
#define TFT_DC    2  // Data Command control pin
#define TFT_RST   4  // Reset pin (could connect to RST pin)
//#define TFT_RST  -1  // Set TFT_RST to -1 if display RESET is connected
to ESP32 board RST

#define TOUCH_CS 0     // Chip select pin (T_CS) of touch screen

Op wo 28 jun 2023 om 12:36 schreef RSZ-Nld ***@***.***>:

**Comment by paulh002** [comment]
I am moving the functions from the old VFO to the newone, for questions you
can use the discussions page on ESPVFO32 repo.

Op wo 28 jun 2023 om 12:36 schreef RSZ-Nld ***@***.***>:

**Comment by RSZ-Nld** [comment]
Ok thanks a lot for youre help !

**Comment by RSZ-Nld** [comment]
Now it works fine, only the touchscreen is again upside down, what do i have to change for that?

**Comment by paulh002** [comment]
void my_touchpad_read(lv_indev_drv_t* indev_driver, lv_indev_data_t* data)
{
uint16_t touchX, touchY;

bool touched = tft.getTouch(&touchX, &touchY, 600);

if (tft.getRotation() == 1)
{
touchX = screenWidth - touchX;
//touchY = screenHeight - touchY;  // only for 3.2 inch
}

if (!touched)
{
data->state = LV_INDEV_STATE_REL;
}
else
{
data->state = LV_INDEV_STATE_PR;

/*Set the coordinates*/
data->point.x = touchX;
data->point.y = touchY;
}
}

remove the comment for touchY

Op wo 28 jun 2023 om 12:55 schreef RSZ-Nld ***@***.***>:

**Comment by RSZ-Nld** [comment]
Yes now it is working, thx ! I am glad that all is working now with Arduino IDE, but V1.8.19 is a problem i gues.

**Comment by paulh002** [comment]
Maybe in the process of downgrading the old library is removed (maybe
cached). I guess when you upgrade to 19 again it will be ok.
But it does not matter so much, if 1 works ok stay on that version.

Op wo 28 jun 2023 om 13:31 schreef RSZ-Nld ***@***.***>:

**Comment by RSZ-Nld** [comment]
Youre right, updated to V1.8.19 and it works !

**Comment by paulh002** [comment]
Thats good!, Currently I am moving the  support for this PCB to ESP32Vfo repository. I adding the TX Power and SWR meter at the moment. You need an SWR board you can find it in the swr-meter repo or use the original from Loftur E. Jonasson.

**Comment by RSZ-Nld** [comment]
Ok nice, take a look at the SWR Meter !
